Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -1126,116 +242,8 @@ theorem lift_iSup_le_lift_iSup' {ι : Type v} {ι' : Type v'} {f : ι → Cardin
     (h : ∀ i, lift.{v'} (f i) ≤ lift.{v} (f' (g i))) : lift.{v'} (iSup f) ≤ lift.{v} (iSup f') :=
   lift_iSup_le_lift_iSup hf hf' h
 
-lemma exists_eq_of_iSup_eq_of_not_isSuccPrelimit
-    {ι : Type u} (f : ι → Cardinal.{v}) (ω : Cardinal.{v})
-    (hω : ¬ IsSuccPrelimit ω)
-    (h : ⨆ i : ι, f i = ω) : ∃ i, f i = ω := by
-  subst h
-  suffices BddAbove (range f) from (isLUB_csSup' this).mem_of_not_isSuccPrelimit hω
-  contrapose! hω with hf
-  rw [iSup, csSup_of_not_bddAbove hf, csSup_empty]
-  exact isSuccPrelimit_bot
-
-lemma exists_eq_of_iSup_eq_of_not_isSuccLimit
-    {ι : Type u} [hι : Nonempty ι] (f : ι → Cardinal.{v}) (hf : BddAbove (range f))
-    {c : Cardinal.{v}} (hc : ¬ IsSuccLimit c)
-    (h : ⨆ i, f i = c) : ∃ i, f i = c := by
-  rw [Cardinal.isSuccLimit_iff] at hc
-  refine (not_and_or.mp hc).elim (fun e ↦ ⟨hι.some, ?_⟩)
-    (Cardinal.exists_eq_of_iSup_eq_of_not_isSuccPrelimit.{u, v} f c · h)
-  cases not_not.mp e
-  rw [← le_zero_iff] at h ⊢
-  exact (le_ciSup hf _).trans h
-
-set_option linter.deprecated false in
-@[deprecated exists_eq_of_iSup_eq_of_not_isSuccLimit (since := "2024-09-17")]
-lemma exists_eq_of_iSup_eq_of_not_isLimit
-    {ι : Type u} [hι : Nonempty ι] (f : ι → Cardinal.{v}) (hf : BddAbove (range f))
-    (ω : Cardinal.{v}) (hω : ¬ ω.IsLimit)
-    (h : ⨆ i : ι, f i = ω) : ∃ i, f i = ω := by
-  refine (not_and_or.mp hω).elim (fun e ↦ ⟨hι.some, ?_⟩)
-    (Cardinal.exists_eq_of_iSup_eq_of_not_isSuccPrelimit.{u, v} f ω · h)
-  cases not_not.mp e
-  rw [← le_zero_iff] at h ⊢
-  exact (le_ciSup hf _).trans h
-
 /-! ### Indexed cardinal `prod` -/
```

Comment:
This section only contains two theorems. Maybe we should re-section this file in some other way?

Reply:
It looks like we can upstream `prod_le_prod` to `Cardinal/Order.lean`, but `prod_eq_of_fintype` requires an import of `Mathlib.Data.Fintype.BigOperators` (this is the only use of that lemma in the file!). We could in fact downstream it to `Cardinal/Finite.lean`, but that would not fit in with the rest of the file (which covers `Nat.card`). For that one, since it does contain `lift`, I'll keep it in `Cardinal/Basic.lean` but move it to the section on `lift`.